challenge_take_out_the_garbage/src/challenge_take_out_the_garbage/take_out.py

```diff
@@ -0,0 +1,191 @@
+# ROS
+import rospy
+import smach
+
+# TU/e
+import robot_skills
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+
+
+class DefaultGrabDesignator(ds.Designator):
+    """ Designator to pick the closest item on top of the table to grab. This is used for testing
+
+    """
+    def __init__(self, robot, surface_designator, area_description, debug=False):
+        """ Constructor
+
+        :param robot: robot object
+        :param surface_designator: designator for the object to grab from
+        :param area_description: string with id of the area where the object should be located in
+        """
+        super(DefaultGrabDesignator, self).__init__(resolve_type=robot_skills.util.entity.Entity)
+
+        self._robot = robot
+        self._surface_designator = surface_designator
+        self.area_description = area_description
+        self._debug = debug
+
+    def resolve(self):
```

Comment:
I think the `resolve()` shouldn't be overwritten, but the `_resolve()` @LoyVanBeek am I right?

(I think you copied this from the storing groceries, so we would need to check all designators around

Reply:
Correct, subclasses should implement `_resolve`